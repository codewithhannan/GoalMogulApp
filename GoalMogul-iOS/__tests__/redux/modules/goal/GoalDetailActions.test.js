import mockStore from '../../../../__mocks__/ReduxMockStore';
import { 
    markGoalAsComplete,
    markNeedAsComplete,
    markStepAsComplete
} from '../../../../src/redux/modules/goal/GoalDetailActions';


// NOTE: avoid using the same mockStore. Always create a new one since the async is not 
// properly added in all the actions
describe("Test goal update actions", () => { 
    let store;
    let getState;
    // Create a new store before each test
    beforeEach(() => {
        store = mockStore({});
        getState = () => ({
            user: { token: '' },
            navigation: { tab: '' }
        });
    });

    it('should dispatch goal update success with complete as true', async () => {
        fetch.mockResponseSuccess(JSON.stringify({}));
        markGoalAsComplete(1, true, 1)(store.dispatch, getState);
        // Wait for change to populate in store. This might get funky
        await new Promise((r) => setTimeout(r, 200));
        expect(store.getActions()).toMatchSnapshot();
    });

    it('should dispatch goal update success with complete as false', async () => {
        fetch.mockResponseSuccess(JSON.stringify({}));
        markGoalAsComplete(1, false, 1)(store.dispatch, getState);
        // Wait for change to populate in store. This might get funky
        await new Promise((r) => setTimeout(r, 200));
        expect(store.getActions()).toMatchSnapshot();
    });
});
