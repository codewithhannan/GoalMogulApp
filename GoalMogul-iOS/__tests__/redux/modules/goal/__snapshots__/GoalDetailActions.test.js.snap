// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test goal update actions should dispatch goal update success with complete as false 1`] = `
Array [
  Object {
    "payload": Object {
      "complete": false,
      "goalId": 1,
      "pageId": 1,
      "tab": "",
      "type": "markGoalAsComplete",
    },
    "type": "goal_detail_update",
  },
  Object {
    "payload": Object {
      "complete": false,
      "goalId": 1,
      "pageId": 1,
      "tab": "",
      "type": "markGoalAsComplete",
    },
    "type": "goal_detail_update_done",
  },
  Object {
    "payload": Object {
      "complete": false,
      "data": undefined,
      "goalId": 1,
      "pageId": 1,
      "tab": "",
    },
    "type": "goal_detail_mark_as_complete_success",
  },
]
`;

exports[`Test goal update actions should dispatch goal update success with complete as true 1`] = `
Array [
  Object {
    "payload": Object {
      "complete": true,
      "goalId": 1,
      "pageId": 1,
      "tab": "",
      "type": "markGoalAsComplete",
    },
    "type": "goal_detail_update",
  },
  Object {
    "payload": Object {
      "complete": true,
      "goalId": 1,
      "pageId": 1,
      "tab": "",
      "type": "markGoalAsComplete",
    },
    "type": "goal_detail_update_done",
  },
  Object {
    "payload": Object {
      "complete": true,
      "data": undefined,
      "goalId": 1,
      "pageId": 1,
      "tab": "",
    },
    "type": "goal_detail_mark_as_complete_success",
  },
]
`;

exports[`hi there should dispatch goal update success with complete as false 1`] = `
Array [
  Object {
    "payload": Object {
      "complete": false,
      "goalId": 1,
      "pageId": 1,
      "tab": "",
      "type": "markGoalAsComplete",
    },
    "type": "goal_detail_update",
  },
  Object {
    "payload": Object {
      "complete": false,
      "goalId": 1,
      "pageId": 1,
      "tab": "",
      "type": "markGoalAsComplete",
    },
    "type": "goal_detail_update_done",
  },
  Object {
    "payload": Object {
      "complete": false,
      "data": undefined,
      "goalId": 1,
      "pageId": 1,
      "tab": "",
    },
    "type": "goal_detail_mark_as_complete_success",
  },
]
`;

exports[`hi there should dispatch goal update success with complete as true 1`] = `
Array [
  Object {
    "payload": Object {
      "complete": true,
      "goalId": 1,
      "pageId": 1,
      "tab": "",
      "type": "markGoalAsComplete",
    },
    "type": "goal_detail_update",
  },
  Object {
    "payload": Object {
      "complete": true,
      "goalId": 1,
      "pageId": 1,
      "tab": "",
      "type": "markGoalAsComplete",
    },
    "type": "goal_detail_update_done",
  },
  Object {
    "payload": Object {
      "complete": true,
      "data": undefined,
      "goalId": 1,
      "pageId": 1,
      "tab": "",
    },
    "type": "goal_detail_mark_as_complete_success",
  },
]
`;
